onboard-server/src/main/java/no/vegvesen/ixn/serviceprovider/OnboardRestController.java

```diff
@@ -368,7 +337,8 @@ public GetDeliveryResponse getDelivery(@PathVariable String serviceProviderName,
 	public void deleteDelivery(@PathVariable String serviceProviderName, @PathVariable String deliveryId) {
 		OnboardMDCUtil.setLogVariables(selfService.getNodeProviderName(), serviceProviderName);
 		this.certService.checkIfCommonNameMatchesNameInApiObject(serviceProviderName);
-		ServiceProvider serviceProvider = checkAndGetServiceProvider(serviceProviderName);
+
+		ServiceProvider serviceProvider = getOrCreateServiceProvider(serviceProviderName);
```

Comment:
Create ServiceProvider here? Also, should cast a NotFoundException if the requested Delivery doesn't exist